Source/ARTRealtime.m

```diff
@@ -404,6 +405,7 @@ - (void)_connect {
         // New connection
         _transport = nil;
     }
+    [self setSuspendTime];
```

Comment:
Maybe it's because I have a pretty bad headache but I can't follow the logic here at all.

What's `suspendTime`? Why are we setting it to the moment the connection is established + `connectionStateTtl`?

Reply:
The suspend time it's the current time + TTL, the time to considerate the connection is suspended. I assigned the suspend time in the moment that the connection starts.

Reply:
May I suggest then `connectedAt`, and then do `if (now > connectedAt + TTL)` to check if we're suspended? That's easier to understand for me anyway. `suspendTime` just tells you it's a time and has to do with suspend.

But anyway... This isn't how it works at all, is it? We're suspended if the time since the last _disconnection_ is greater than the connection state TTL.

I'd suggest an alternative solution: before moving to CONNECTING, store whether we are coming from DISCONNECTED or SUSPENDED and, on connection failure, move back to that state (the move from DISCONNECTED to SUSPENDED already works). As simple as that, we don't need to deal with more deadlines if I'm understanding things correctly.

Reply:
On second thought, the new way of moving from DISCONNECTED to SUSPENDED is probably clearer that what we had before. So, maybe just fix this:

> But anyway... This isn't how it works at all, is it? We're suspended if the time since the last _disconnection_ is greater than the connection state TTL.

(So instead of `connectedAt` it could be `disconnectedAt` or `connectionLostAt` or something like that.)

Reply:
@tcard I understand your point. I think `connectionLostAt` is a better name because I should only set the time on the first disconnection. I need some sort of logic to identify the connection lost, something like:

```
if (stateChange.previous == ARTRealtimeConnected || stateChange.current == ARTRealtimeDisconnected) {
    [self setConnectionLostTime];
}
```

Even so, I think it's simpler setting the expected suspension time right in the beginning like it's done right now. Maybe the naming is not the best, maybe `setExpectedSuspensionTime` instead of `setSuspendTime` would be better?

Reply:
@tcard Does b600118 turn things more clear?

Reply:
@ricardopereira 

> Even so, I think it's simpler setting the expected suspension time right in the beginning like it's done right now. 

My point is that this isn't how it should work. The suspension timer starts when the connection is _broken_, not when it's _established_. You're calling `setConnectionLostTime` when the connection is established, while it should be called when it's broken.

What you're doing now implies that, if we connect, then let more than 2 minutes (the connection state TTL) pass, then the connection is broken, we go straight to SUSPEND. What should happen is that we go to DISCONNECTED for 2 minutes, then to SUSPEND, now matter how long we were connected.

Reply:
(But I'm OK with `setExpectedSuspensionTime` as a name otherwise. I still think storing `connectionLostAt` or similar is more clear, but that's a subjective thing.)

Reply:
@tcard Got it, please look at 26baa96. Thanks for your explanation and patience.